packages/taquito/src/contract/prepare.ts

```diff
@@ -339,3 +341,32 @@ export const createSmartRollupOriginateOperation = async ({
     parameters_ty: parametersType,
   } as RPCSmartRollupOriginateOperation;
 };
+
+export const createStakingOperation = async (
+  {
+    amount,
+    fee = DEFAULT_FEE.TRANSFER,
+    gasLimit = DEFAULT_GAS_LIMIT.TRANSFER,
+    storageLimit = DEFAULT_STORAGE_LIMIT.TRANSFER,
+    mutez = false,
+  }: StakingParams,
+  pkh: string,
+  entrypoint: StakingEntrypoint
+) => {
+  const operation: RPCTransferOperation = {
+    source: pkh,
+    destination: pkh,
+    kind: OpKind.TRANSACTION,
+    fee,
+    gas_limit: gasLimit,
+    storage_limit: storageLimit,
+    amount: mutez ? amount.toString() : format('tz', 'mutez', amount).toString(),
+    parameters: {
+      entrypoint,
+      value: {
+        prim: 'Unit',
```

Comment:
Would like to not hard code a generalized `createStakingOperation` value: {prim: 'Unit'} here, should happened in their individual function. As hardcoded to unit only workable with current ops supported, we need to leave room for more pseudo ops like set_delegate_parameters to pass value in

Reply:
That is more code, and if that is ever needed, then we can move it to the individual functions

Reply:
We can make it scalable now, would like to understand what's the reason to postpone?
Op set_delegate_parameters do need to pass param value in, even we don't look to support it in current release. It's a fact tezos have intention to pass param value with pseudo operations.

Reply:
Maybe it's a matter of taste, but the common principle is [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). Short for "You aren't gonna need it", which states that "a programmer should not add functionality until deemed necessary".

Reply:
c.c. @dsawali